### ISSUE TITLE
Refactor etcd.Deploy method as its too long and does too many things

### ISSUE THREAD

**Issue by unmarshall** [gardener/gardener#5977]
**How to categorize this issue?**
/area control-plane
/area quality
/kind enhancement
/kind technical-debt

**What would you like to be added**:
In `operation/botanist/etcd/etcd.go` method: `func (e *etcd) Deploy(ctx context.Context) error` has significantly grown in size especially after the [PR#5741](https://github.com/gardener/gardener/pull/5741/).
The method creates a LOT of resources which are sequentially created one after another. This makes code difficult to read, difficult to test. Additional resources will keep getting added making this method larger and larger. 

**Why is this needed**:
An alternate way to do this would be to break the down this method into multiple smaller functions where each function creates a single resource and then sequence the resource creation steps via `Flow` framework.
There are numerous places in the gardener code where `Flow` is already used. As an example one can refer to `gardener/pkg/gardenlet/controller/shoot/shoot_control_reconcile.go`

**Comment by timebertt** [comment]
The linked PR is not even merged yet. So do we have an issue for something that's not even on master and can be tackled right away?

**Comment by unmarshall** [comment]
It was suggested that we create another issue for taking care of refactoring. [#5741](https://github.com/gardener/gardener/pull/5741/) is time sensitive and needs to go in first. There is an ongoing effort to even split this PR into 3 smaller PR's, hopefully allowing merging of multi-node specific changes. We wish to tackle the refactoring effort separately as it is not a must to have for multi-node/multi-zone.

**Comment by timebertt** [comment]
TBH, I don't like to consciously add technical debt, that can be avoided in the first place. This issue is clearly a big red sign for such a case.

**Comment by unmarshall** [comment]
I completely agree with you that we should not create new technical debt when the new changes are not yet merged to master which is why i left a review comment in [PR#5471](https://github.com/gardener/gardener/pull/5741/) only to be told 2 things:
- A method which after the new changes has grown to ~560 lines of code is not an immediate concern
- This should be handled as a separate PR.

In the end i want this to be changed as i follow the principle of smaller functions which do one thing only making it easier to reason about code. However this is always a point of contention as each developer has their own preference which is why i see different priorities being assigned to such changes.

Please feel free to also comment in the PR and block the merge of the PR and then you can close this issue. OR if you think that having such large methods are not a concern then also you can close this PR.

**Comment by timuthy** [comment]
> TBH, I don't like to consciously add technical debt, that can be avoided in the first place. This issue is clearly a big red sign for such a case.

+1

Please don't get me wrong but I see that such issues are typical candidates which get auto-closed after `90d` because no one will have capacity to work on it. If the refactoring was a prerequisite for #5741, then we should have done it first or along the way to a certain extent like **Boy Scout Rule** suggests. So in the meantime this is probably just going to be yet another open issue that is not going to be resolved.

**Comment by unmarshall** [comment]
@timuthy Agreed. 
>If the refactoring was a prerequisite for https://github.com/gardener/gardener/pull/5471, then we should have done it first or along the way to a certain extent like Boy Scout Rule suggests.

Well this was a review comment that i added. I do not know how the code ownership works in gardener yet, but i assume the final decision to refactor or not is not mine (at least in gardener core), therefore it was an observation made and recorded as part of the PR. Please take a call and if you decide that the review comment needs to be incorporated as part of the PR, then please make that explicit in the PR by adding your own comment :)

My intent is to get that refactoring done - does not matter if it is done as part of the PR or via a separate PR.

**Comment by timebertt** [comment]
> My intent is to get that refactoring done - does not matter if it is done as part of the PR or via a separate PR.

2 months later after https://github.com/gardener/gardener/pull/5741/ has been merged, the issue is still open and the code is not refactored...
Seems like @timuthy was right after all:

> Please don't get me wrong but I see that such issues are typical candidates which get auto-closed after `90d` because no one will have capacity to work on it. If the refactoring was a prerequisite for #5741, then we should have done it first or along the way to a certain extent like **Boy Scout Rule** suggests. So in the meantime this is probably just going to be yet another open issue that is not going to be resolved.

**Comment by unmarshall** [comment]
> Seems like @timuthy was right after all

Well i waited for response for https://github.com/gardener/gardener/issues/5977#issuecomment-1129795851 
and then finally gave up. I see a lot of code with extremely long functions which do more than one thing all over the code base. Therefore i asked for views from core contributors to gardener core if they think it is ok to have this or it should be changed. I assumed that this is not a serious concern and i backed out.

I can still change it if you or any core contributor agrees that it needs to change. Shreyas is making some more changes to incorporate changes suggested in GEP-20. I can taken over and do the refactoring once he is done.
